lib/platform/github/gh-graphql-wrapper.js

```diff
@@ -0,0 +1,111 @@
+const get = require('./gh-got-wrapper');
+
+let repoOwner = '';
+let repoName = '';
+let repoItem = '';
+let repoItemFilterBy = '';
+let repoItemNodeList = [];
+let repoResultNumEls = 100;
+
+const graphql = {};
+
+graphql.setRepoOwner = owner => {
+  repoOwner = owner;
+};
+
+graphql.setRepoName = name => {
+  repoName = name;
+};
+
+graphql.setRepoItem = item => {
+  repoItem = item;
+};
+
+graphql.setRepoItemFilterBy = itemFilterBy => {
+  repoItemFilterBy = itemFilterBy;
+};
+
+graphql.setRepoItemNodeList = itemNodeList => {
+  repoItemNodeList = itemNodeList;
+};
+
+graphql.setRepoResultNumEls = numEls => {
+  repoResultNumEls = numEls;
+};
+
+// returns [successStatus, repo.item.nodes, nextCursor]
+graphql.get = async (cursor = '') => {
+  const url = 'graphql';
+  const headers = {
+    accept: 'application/vnd.github.merge-info-preview+json',
+  };
+  const afterCursor = cursor === '' ? null : `"${cursor}"`;
+  // prettier-ignore
+  const nodeListString = repoItemNodeList.join('\n');
+  const query = `
+  query {
+    repository(owner: "${repoOwner}", name: "${repoName}") {
+      ${repoItem}(first: ${repoResultNumEls}, after:${afterCursor}, orderBy: {field: UPDATED_AT, direction: DESC}, filterBy: ${repoItemFilterBy}) {
+        pageInfo {
+          startCursor
+          hasNextPage
+        }
+        nodes {
+          ${nodeListString}
+        }
+      }
+    }
+  }
+  `;
+
+  const options = {
+    headers,
+    body: JSON.stringify({ query }),
+    json: false,
+  };
+
+  try {
+    const res = JSON.parse((await get.post(url, options)).body);
+
+    if (!res.data) {
+      // retry query until numElements == 1
+      if (repoResultNumEls === 1) {
+        logger.info({ query, res }, 'No graphql res.data');
+        return [false, [], ''];
+      }
+
+      repoResultNumEls = parseInt(repoResultNumEls / 2);
```

Comment:
Because you have declared this variable in the file's top level scope, won't that mean the number is shared by *all* concurrent queries? And does it ever get reset back to 100? I think you need to include it in the function's query parameters so that you can reduce it recursively.